### PR TITLE
Added new icon `InterestOfBalance` in size 24.

### PR DESCRIPTION
**Pull request by giedoka**
Addresses #ticket-number.

## Purpose

Added new icon `InterestOfBalance` in size 24.

## Approach and changes

Exported and optimized the svgs from [Figma](https://www.figma.com/file/LnhSTiHtYMVw3Ppw5QZVph/Circuit-UI-Foundation?type=design&node-id=17083-139&mode=design&t=hXCOyDf9sK6PJG7b-4).
Updated the manifest.
Created a changeset.

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: packages/icons/manifest.json [hunk 1]
```diff
@@ -986,6 +986,11 @@
       "category": "Finance",
       "size": "24"
     },
+    {
+      "name": "interest_of_balance",
```
Comment (Burtchen) [inline packages/icons/manifest.json:990]: What is 'interest of balance' supposed to be? The[Google Search for it](https://www.google.com/search?q=%22interest+of+balance%22) does not yield anything plausible

Comment (a5e) [inline packages/icons/manifest.json:990]:
```suggestion
      "name": "interest_on_balance",
```

File: packages/icons/web/v2/interest_of_balance_24.svg
Comment (a5e) [inline packages/icons/web/v2/interest_of_balance_24.svg:1]: Rename the file to `interest_on_balance`

Comment (giedoka) [reply packages/icons/web/v2/interest_of_balance_24.svg:1]: thank you, already fixed after @Burtchen CR ;)

File: packages/icons/manifest.json (hunk 1 above)
Comment (giedoka) [reply packages/icons/manifest.json]: oops... Thank you for pointing the typo 😅

**Review by a5e — APPROVED** [review]

**Review by Burtchen — APPROVED** [review]
📊

**Pull request merged**